File: library/src/main/java/com/afollestad/assent/AssentFragment.java
package com.afollestad.assent;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v13.app.FragmentCompat;

/**
 * @author Aidan Follestad (afollestad)
 */
public class AssentFragment extends Fragment implements FragmentCompat.OnRequestPermissionsResultCallback {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Assent.setFragment(this, this);
    }

    @Override
    public void onResume() {
        super.onResume();
        Assent.setFragment(this, this);
    }

    @Override
    public void onPause() {
        super.onPause();
        if (getActivity() != null && getActivity().isFinishing())
            Assent.setFragment(this, null);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        AssentBase.LOG("AssentFragment", "onRequestPermissionsResult(): %d, %s, %s",
                requestCode, AssentBase.join(permissions), AssentBase.join(grantResults));
        Assent.handleResult(permissions, grantResults);
    }
}

File: library/src/main/java/com/afollestad/assent/AssentFragment.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
@file:Suppress("unused")

package com.afollestad.assent

import android.os.Bundle
import android.support.v4.app.Fragment
import com.afollestad.assent.Assent.Companion.onPermissionsResponse
import com.afollestad.assent.Assent.Companion.setAssentFragment

/** @author Aidan Follestad (afollestad) */
open class AssentFragment : Fragment() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setAssentFragment(this, this)
  }

  override fun onResume() {
    super.onResume()
    setAssentFragment(this, this)
  }

  override fun onPause() {
    setAssentFragment(this, null)
    super.onPause()
  }

  override fun onRequestPermissionsResult(
    requestCode: Int,
    permissions: Array<out String>,
    grantResults: IntArray
  ) {
    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    onPermissionsResponse(
        permissions = permissions,
        grantResults = grantResults
    )
  }
}
